src/model/rfdn.py

```diff
@@ -1,32 +1,39 @@
 import torch
 import torch.nn as nn
 import model.block as B
+from model import common
 
 def make_model(args, parent=False):
-    model = RFDN()
+    model = RFDN(args)
     return model
 
 
 class RFDN(nn.Module):
-    def __init__(self, in_nc=3, nf=50, num_modules=4, out_nc=3, upscale=4):
+    def __init__(self, args):
         super(RFDN, self).__init__()
 
-        self.fea_conv = B.conv_layer(in_nc, nf, kernel_size=3)
+        if args.n_feats != 50:
+            print(f'WARNING: Using non paper num output channels of {args.n_feats} instead of 50.')
+        self.fea_conv = B.conv_layer(args.n_colors, args.n_feats, kernel_size=3)
 
-        self.B1 = B.RFDB(in_channels=nf)
-        self.B2 = B.RFDB(in_channels=nf)
-        self.B3 = B.RFDB(in_channels=nf)
-        self.B4 = B.RFDB(in_channels=nf)
-        self.c = B.conv_block(nf * num_modules, nf, kernel_size=1, act_type='lrelu')
+        self.B1 = B.RFDB(in_channels=args.n_feats)
+        self.B2 = B.RFDB(in_channels=args.n_feats)
+        self.B3 = B.RFDB(in_channels=args.n_feats)
+        self.B4 = B.RFDB(in_channels=args.n_feats)
+        self.c = B.conv_block(args.n_feats * args.num_rfdb_blocks, args.n_feats, kernel_size=1, act_type='lrelu')
```

Comment:
Shouldn't num_rfdb_blocks be equal to the number of actually existing RFDB blocks (in this case 4, see lines 19-22) instead of allowing to pass another value here. Otherwise, the number of channels wouldn't match, no?